### PR TITLE
Add check to enable gems if o3de_project is set

### PR DESCRIPTION
**Pull request by amzn-changml**
## What does this PR do?

Adds a condition for the enable gem step in AR to only trigger if the o3de_projects variable is set. Previously it ran automatedtesting, then looped into empty projects

## How was this PR tested?

Tested in a sandbox Jenkins

### CODE REVIEW

**Review by lumberyard-employee-dm — APPROVED** [review]

File: scripts/build/Jenkins/Jenkinsfile [hunk 1]
```diff
@@ -483,8 +483,10 @@ def ExecuteControlFile(controlFile, repoName, processed_repos, o3de_project, def
     //queue enable all gems specified in the control file
     if(controlFile.ENABLE_GEMS) {
         controlFile.ENABLE_GEMS.each { gemName ->
-            defer_enable_gems.add([COMMAND: "${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} enable-gem --gem-path ${workspace}/${repoName}/${gemName} --project-name ${o3de_project} || echo>nul", ECHO: "Enabling ${workspace}/${repoName}/${gemName} for ${o3de_project}"])
-        }
+            if(o3de_project) {
+                defer_enable_gems.add([COMMAND: "${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} enable-gem --gem-path ${workspace}/${repoName}/${gemName} --project-name ${o3de_project} || echo>nul", ECHO: "Enabling ${workspace}/${repoName}/${gemName} for ${o3de_project}"])
+            }
+        }
```
Comment (lumberyard-employee-dm) [inline scripts/build/Jenkins/Jenkinsfile:489]: Should we place the check for the `o3de_project` variable outside of the foreach loop.

There is no reason to iterate over the gem names if the project name isn't set at all

Comment (amzn-changml) [reply scripts/build/Jenkins/Jenkinsfile:489]: I think putting it outside of the loop would only check if the list is empty; the issue is that there may be still be elements with project names along with empty ones. We could use a `-` operator to remove all null and blank elements, though. Will try that in sandbox

**Review by brianherrera — APPROVED** [review]

File: scripts/build/Jenkins/Jenkinsfile (hunk 1 above)
Comment (amzn-changml) [reply scripts/build/Jenkins/Jenkinsfile]: Unfortunately, it didn't make a difference. It also ran twice, even after removing the null elements. It's as though list length is being retained somehow. May be a quirk of Java/groovy that needs further investigation

File: scripts/build/Jenkins/Jenkinsfile [hunk 2]
```diff
@@ -481,12 +485,12 @@ def ExecuteControlFile(controlFile, repoName, processed_repos, o3de_project, def
         }
     }
     //queue enable all gems specified in the control file
-    if(controlFile.ENABLE_GEMS) {
+    if(controlFile.ENABLE_GEMS && o3de_project) {
         controlFile.ENABLE_GEMS.each { gemName ->
             if(o3de_project) {
-                defer_enable_gems.add([COMMAND: "${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} enable-gem --gem-path ${workspace}/${repoName}/${gemName} --project-name ${o3de_project} || echo>nul", ECHO: "Enabling ${workspace}/${repoName}/${gemName} for ${o3de_project}"])
+                defer_enable_gems.add([COMMAND: "${workspace}/${ENGINE_REPOSITORY_NAME}/scripts/o3de${ext} enable-gem --gem-path ${workspace}/${repoName}/${gemName} --project-name ${o3de_project}", ECHO: "Enabling ${workspace}/${repoName}/${gemName} for ${o3de_project}"])
```
Comment (lumberyard-employee-dm) [inline scripts/build/Jenkins/Jenkinsfile:491]: The inner `if(o3de_project)` check should no longer be needed.

Comment (amzn-changml) [reply scripts/build/Jenkins/Jenkinsfile:491]: As noted in the previous comment, removing the inner check results in more runs (OpenVk and OpenXR running twice)

**Pull request merged**